Sources/Turf/Geometries/LineString.swift

```diff
@@ -78,9 +78,65 @@ extension LineString {
     }
     
     /**
-     Returns the portion of the line string that begins at the given coordinate and extends the given distance along the line string.
+     Returns the portion of the line string that begins at the given start distance and extends the given stop distance along the line string.
      
-     This method is roughly equivalent to the [turf-line-slice-along](https://turfjs.org/docs/#lineSliceAlong) package of Turf.js ([source code](https://github.com/Turfjs/turf/tree/master/packages/turf-line-slice-along/)), except that it accepts a starting position instead of a starting distance along the line string.
+     This method is equivalent to the [turf-line-slice-along](https://turfjs.org/docs/#lineSliceAlong) package of Turf.js ([source code](https://github.com/Turfjs/turf/tree/master/packages/turf-line-slice-along/)).
+     */
+    public func trimmed(from startDistance: LocationDistance, to stopDistance: LocationDistance) -> LineString? {
+        // The method is porting from https://github.com/Turfjs/turf/blob/5375941072b90d489389db22b43bfe809d5e451e/packages/turf-line-slice-along/index.js
+        guard startDistance >= 0.0 && stopDistance >= startDistance else { return nil }
+        let coordinates = self.coordinates
+        var traveled: LocationDistance = 0
+        var slice = [LocationCoordinate2D]()
+        
+        for i in 0..<coordinates.endIndex {
+            if startDistance >= traveled && i == coordinates.endIndex - 1 {
+                break
+            } else if traveled > startDistance && slice.isEmpty {
+                let overshoot = startDistance - traveled
+                if overshoot == 0.0 {
+                    slice.append(coordinates[i])
+                    return LineString(slice)
+                }
```

Comment:
This looks very similar to `indexedCoordinateFromStart(distance:)`, which was ported from turf-along. Is there an opportunity to reuse that method?

https://github.com/mapbox/turf-swift/blob/33da423c56216c4dc41c68beb130019081085113/Sources/Turf/Geometries/LineString.swift#L162-L179

Reply:
Theoretically, we could using the following the avoid the code duplication:
```
guard startDistance >= 0.0 && stopDistance >= startDistance else { return nil }
guard let startIndexedCoordinate = indexedCoordinateFromStart(distance: startDistance) else { return nil }
return trimmed(from: startIndexedCoordinate.coordinate, distance: stopDistance - startDistance)
```
But in fact, we shouldn't do that. The above failed several unit tests with the following reasons:
1. When the startDistance is larger than the lineString distance, it should return nil but in fact it would return the last coordinate.
2. The `trimmed(from:distance:)` doesn't accept the `startIndexedCoordinate.coordinate` directly, but in fact it would call `closestCoordinate(to:)` again to find the coordinate. It would lead to the wrong start coordinate and possible wrong end coordinate, when the stop distance is smaller than the whole distance of LineString.

Except the test failure, it would also increase the calculation, because the `closestCoordinate(to:)` would loop through the lineString's coordinates for one more time.